tests/view/test_slice.cpp

```diff
@@ -0,0 +1,119 @@
+TEST(view, slice_rank_1)
+{
+    const Kokkos::View<double[dim_1], execution_space> view_1_static("rank-1 view with static extent");
+    const Kokkos::View<double*      , execution_space> view_1_dynami("rank-1 view with dynamic extent", dim_1);
+
+    CHECK_THAT(utils::view::slice<1>(view_1_static), 1, Kokkos::Array{dim_1});
+    CHECK_THAT(utils::view::slice<1>(view_1_dynami), 1, Kokkos::Array{dim_1});
+
+    CHECK_THAT(utils::view::slice(view_1_static, 0), 0, (Kokkos::Array<size_t, 0>{}));
+    CHECK_THAT(utils::view::slice(view_1_dynami, 0), 0, (Kokkos::Array<size_t, 0>{}));
+}
+
+//! @test Check that @ref view::slice works as expected for a rank-2 @c Kokkos::View.
+TEST(view, slice_rank_2)
+{
+    const Kokkos::View<double[dim_1][dim_2], execution_space> view_2_static("rank-2 view with static extents");
+    const Kokkos::View<double**            , execution_space> view_2_dynami("rank-2 view with dynamic extents", dim_1, dim_2);
+    const Kokkos::View<double*[dim_2]      , execution_space> view_2_mixed ("rank-2 view with mixed extents"  , dim_1);
+
+    CHECK_THAT(utils::view::slice<2>(view_2_static, 0), 1, Kokkos::Array{dim_2});
+    CHECK_THAT(utils::view::slice<2>(view_2_dynami, 0), 1, Kokkos::Array{dim_2});
+    CHECK_THAT(utils::view::slice<2>(view_2_mixed , 0), 1, Kokkos::Array{dim_2});
+}
+
+//! @test Check that @ref view::slice works as expected for a rank-3 @c Kokkos::View.
+TEST(view, slice_rank_3)
+{
+    const Kokkos::View<double[dim_1][dim_2][dim_3], execution_space> view_3_static("rank-3 view of static extents");
+    const Kokkos::View<double***                  , execution_space> view_3_dynami("rank-3 view of dynamic extents", dim_1, dim_2, dim_3);
+    const Kokkos::View<double*[dim_2][dim_3]      , execution_space> view_3_mixed ("rank-3 view of mixed extents"  , dim_1);
+
+    CHECK_THAT(utils::view::slice<3>(view_3_static, 0), 2, (Kokkos::Array{dim_2, dim_3}));
+    CHECK_THAT(utils::view::slice<3>(view_3_dynami, 0), 2, (Kokkos::Array{dim_2, dim_3}));
+    CHECK_THAT(utils::view::slice<3>(view_3_mixed , 0), 2, (Kokkos::Array{dim_2, dim_3}));
+
+    CHECK_THAT(utils::view::slice<3>(view_3_mixed , 0, Kokkos::ALL), 2, (Kokkos::Array{dim_2, dim_3}));
+}
+
+//! @test Check that @ref view::slice works as expected for a rank-4 @c Kokkos::View.
+TEST(view, slice_rank_4)
+{
+    const Kokkos::View<double[dim_1][dim_2][dim_3][dim_4], execution_space> view_4_static("rank-4 view of static extents");
+
+    CHECK_THAT(utils::view::slice<4>(view_4_static, 0, Kokkos::ALL), 3, (Kokkos::Array{dim_2, dim_3, dim_4}));
+
+    CHECK_THAT(utils::view::slice(view_4_static, 0, 0, 0, 0), 0, (Kokkos::Array<size_t, 0>{}));
+
+    CHECK_THAT(utils::view::slice<4>(view_4_static, Kokkos::make_pair(0, 2)), 4, (Kokkos::Array<size_t, 4>{2, dim_2, dim_3, dim_4}));
+}
+
+template <concepts::ViewOfRank<4> ViewType>
+bool slice_on_device(const ViewType& view, const size_t expected_size)
```

Comment:
Could also become `test_slice_on_device` with a convention to start such functions with `test_`? May be worth checking in `Kokkos` how such functions are named.

Reply:
Suggestion applied, haven't checked on the `Kokkos` side though.